### ISSUE TITLE
How to persist layout between page changes? thank you very much

### ISSUE THREAD

**Issue by Orecic** [zeit/next.js#4166]
- [ ] I have searched the [issues](https://github.com/zeit/next.js/issues?q=is%3Aissue) of this repository and believe that this is not a duplicate.

## Steps to Reproduce (for bugs)

1.
2.
3.
4.

## Your Environment

| Tech    | Version |
|---------|---------|
| next    |         |
| node    |         |
| OS      |         |
| browser |         |
| etc     |         |

**Comment by timneutkens** [comment]
Duplicate of #4145

**Comment by Orecic** [comment]
@timneutkens hello, I still don't understand how to persist layout between page changes, Can you give me an example? thank you very much

**Comment by timneutkens** [comment]
https://github.com/zeit/next.js/issues/88

**Comment by timneutkens** [comment]
If you want to not re-render that's not possible yet.
But:

https://github.com/zeit/next.js/tree/canary/examples/layout-component

**Comment by Orecic** [comment]
https://github.com/zeit/next.js/tree/canary/examples/layout-component , this example does not work, it cause re-render. Will support golbal layout  in the next version? thanks.

**Comment by Orecic** [comment]
@timneutkens  hi , https://github.com/zeit/next.js/tree/canary/examples/layout-component , this example does not work, it cause re-render. Will support golbal layout in the next version? thanks.

**Comment by timneutkens** [comment]
I believe I linked you to 2 issues explaining exactly that. It’ll be possible with _app.js in next 6. Currently out on canary.

**Comment by pozylon** [comment]
@Orecic This example will show you what you're looking for, it will hopefully get into canary soon: https://github.com/zeit/next.js/pull/4177 cc @timneutkens

**Comment by Orecic** [comment]
@timneutkens @pozylon thank you very much.  I have  another question，I have a golbal nav，when page A link to page B,  page B rendered, but when link back to page A, page A re-rendered ,so page A's getInitialProps execute twice， I want page A's getInitialProps execute only once, what should I do
？ thank you.

**Comment by Orecic** [comment]
@pozylon  when switching routes ,getInitialProps execute only once, Can it be achieved?
